app/channel.go

```diff
@@ -1600,30 +1600,31 @@ func (a *App) MarkChannelsAsViewed(channelIds []string, userId string, clearPush
 	channelsToClearPushNotifications := []string{}
 	if *a.Config().EmailSettings.SendPushNotifications && clearPushNotifications {
 		for _, channelId := range channelIds {
-			if model.IsValidId(channelId) {
-				result := <-a.Srv.Store.Channel().GetMember(channelId, userId)
-				if result.Err != nil {
-					mlog.Warn(fmt.Sprintf("Failed to get membership %v", result.Err))
-					continue
-				}
-				member := result.Data.(*model.ChannelMember)
+			if !model.IsValidId(channelId) {
+				return nil, model.NewAppError("App.MarkChannelsAsViewed", "Bad channelId.", nil, "channelId="+channelId, http.StatusBadRequest)
```

Comment:
In this case, I'd prefer to just skip the invalid ID with a warning because I think the case of "only some of those channels were marked as read" is acceptable here. If it were something where we only wanted to mark one channel as read if we could mark all of them as read, I'd say we should fail early, but I don't believe that's the case here.

Reply:
Ya, I don't think this is too much of a problem because that kind if failure is probably programmer error. But I don't think this was the place to validate that anyway. Modified so the values are checked in the api4 layer.